goft.go

```diff
@@ -1,24 +1,73 @@
-package goft
-
-import "fmt"
-
-type vector []float32
-type matrix [][]float32
-
-// multiply two vectors of the same size together
-func (v1 vector) multiply(v2 vector) (vector, error) {
-
-	v1Len := len(v1)
-	v2Len := len(v2)
-
-	if v1Len != v2Len {
-		return nil, fmt.Errorf("vectors aren't the same size: v1 size = %d, v2 size = %d", v1Len, v2Len)
-	}
-
-	v3 := make(vector, v1Len)
-	for i := 0; i < v1Len; i++ {
-		v3[i] = v1[i] * v2[i]
-	}
-
-	return v3, nil
-}
+package goft
+
+import "fmt"
+
+type vector []float32
+
+type Dimension struct {
+	rows int
+	cols int
+}
+
+type Matrix struct {
+	data []float32
+	dim  Dimension
+}
+
+func (d1 *Dimension) Equals(d2 Dimension) bool {
+	return (d1.rows == d2.rows) && (d1.cols == d2.cols)
+}
+
+func (m *Matrix) unsafeAt(row, col int) *float32 {
+	return &m.data[row*m.dim.cols+col]
+}
+
+// multiply two vectors of the same size together
+func (v1 vector) multiply(v2 vector) (vector, error) {
+
+	v1Len := len(v1)
+	v2Len := len(v2)
+
+	if v1Len != v2Len {
+		return nil, fmt.Errorf("vectors aren't the same size: v1 size = %d, v2 size = %d", v1Len, v2Len)
+	}
+
+	v3 := make(vector, v1Len)
+	for i := 0; i < v1Len; i++ {
+		v3[i] = v1[i] * v2[i]
+	}
+
+	return v3, nil
+}
+
+func ZeroMat(rows, cols int) Matrix {
+	mData := make([]float32, rows*cols)
+	mDim := Dimension{rows: rows, cols: cols}
+	return Matrix{data: mData, dim: mDim}
+}
+
+func Multiply(a, b Matrix) (Matrix, error) {
```

Comment:
Change to a method, that is consistent with the vector multiply - 
`func (m1 Matrix) Multiply(m2 Matrix) (Matrix, error) {}`

Reply:
DOne